crates/flashblocks-rpc/src/state.rs

```diff
@@ -64,6 +64,35 @@ where
         }
     }
 
+    pub fn clear(&self) {
+        if let Some(prev) = self.pending_block.swap(None) {
+            self.metrics.pending_clear_catchup.increment(1);
+            self.metrics
+                .pending_snapshot_height
+                .set(prev.block_number() as f64);
+            self.metrics
+                .pending_snapshot_fb_index
+                .set(prev.flashblock_index() as f64);
+        }
+    }
+
+    pub fn clear_on_canonical_catchup(&self, canonical_number: u64) {
```

Comment:
Can we call this on_canonical_block_received and update the params to take the full block, this:

- Is consistent with the Flashblocks [method](https://github.com/base/node-reth/blob/main/crates/flashblocks-rpc/src/state.rs#L324)
- We're likely going to extend this method to do more than clear to fix https://github.com/base/node-reth/issues/59


It may